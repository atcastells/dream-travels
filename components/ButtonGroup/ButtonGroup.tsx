import classNames from "classnames";
import { useEffect, useMemo, useState } from "react";
import { Button } from "../Button/Button";

interface ButtonGroupProps {
  buttonText: React.ReactNode[];
  onSelect?: (index: number) => void;
}

export const ButtonGroup = ({ buttonText, onSelect }: ButtonGroupProps) => {
  const [selected, setSelected] = useState(0);

  useEffect(() => {
    onSelect?.(selected);
  }, [selected]);

  const buttons = useMemo(() => {
    return buttonText.map((text, index) => {
      const isFirst = index === 0;
      const isLast = index === buttonText.length - 1;
      return (
        <Button
          key={index}
          className={classNames(
            "py-2 border-grey-200 border",
            isFirst ? "rounded-l-full rounded-r-none" : "",
            isLast ? "rounded-r-full rounded-l-none" : "",
            !isFirst && !isLast ? "rounded-none" : "",
            selected === index ? "bg-gray-100" : ""
          )}
          onClick={() => {
            setSelected(index);
          }}
        >
          {text}
        </Button>
      );
    });
  }, [buttonText, selected]);

  return (
    <div className="flex flex-row items-center justify-center my-4">
      {buttons}
    </div>
  );
};
