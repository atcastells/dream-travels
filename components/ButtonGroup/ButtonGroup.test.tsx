import { expect, jest, test } from "@jest/globals";
import { render, screen } from "@testing-library/react";

import { ButtonGroup } from "./ButtonGroup";

test("should render buttons", () => {
  const buttonText = ["All", "Upcoming", "Completed"];

  const onSelectSpy = jest.fn();

  render(
    <ButtonGroup onSelect={(index) => onSelectSpy()} buttonText={buttonText} />
  );

  const allButton = screen.getByText("All");
  const upcomingButton = screen.getByText("Upcoming");
  const completedButton = screen.getByText("Completed");

  expect(allButton).toBeTruthy();
  expect(upcomingButton).toBeTruthy();
  expect(completedButton).toBeTruthy();

  //Click on the first button
  allButton.click();

  expect(onSelectSpy).toHaveBeenCalledTimes(1);

  //Snapshot testing
  expect(allButton).toMatchSnapshot();
});
