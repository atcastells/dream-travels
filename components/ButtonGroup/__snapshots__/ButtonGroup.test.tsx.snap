// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render buttons 1`] = `
<button
  class="py-2 border-grey-200 border rounded-l-full rounded-r-none bg-gray-100 rounded-full bg-white text-black px-2"
>
  All
</button>
`;
